src/border.jl

```diff
@@ -135,13 +260,170 @@ end
+add.
+
+# Output
+
+An expansion of the input image in which additional pixels are derived
+from the border of the input image using the extrapolation scheme specified by
+`border`.
+
+# Details
+
+You can specify the element type `T` of the output image.
+
+## Options
+Valid `border` options are described below.
+
+### `Pad`
+The type `Pad` designates the form of padding which should be used to
+extrapolate pixels beyond the boundary of an image. Instances must set `style`,
+a Symbol specifying the boundary conditions of the image.
+Symbol must be on one of:
+- `:replicate` (repeat edge values to infinity),
+- `:circular` (image edges "wrap around"),
+- `:symmetric` (the image reflects relative to a position between pixels),
+- `:reflect` (the image reflects relative to the edge itself).
+
+Refer to the documentation of [`Pad`](@ref) for more details and examples for
+each option.
+
+### `Fill`
+The type `Fill` designates a particular value which will be used to
+extrapolate pixels beyond the boundary of an image. Refer to the documentation
+of [`Fill`](@ref) for more details and illustrations.
+
+# Examples
+Each example is based on the input array
+```math
+\\mathbf{A} =
+\\boxed{
+\\begin{matrix}
+ 1  & 2  &  3  &  4 & 5  & 6 \\\\
+ 2  & 4  &  6  &  8 & 10 & 12 \\\\
+ 3  & 6  &  9  & 12 & 15 & 18 \\\\
+ 4  & 8  & 12  & 16 & 20 & 24 \\\\
+ 5  & 10 & 15  & 20 & 25 & 30 \\\\
+ 6  & 12 & 18  & 24 & 30 & 36
+ \\end{matrix}}.
+```
+## Examples with `Pad`
+The command `padarray(A, Pad(:replicate,4,4))` yields
+```math
+\\boxed{
+\\begin{array}{ccccccccccccc}
+1 & 1 & 1 & 1 & 1  &  2  &  3  &  4  &  5  &  6  &  6  &  6  &  6  &  6 \\\\
+1 & 1 & 1 & 1 & 1  &  2  &  3  &  4  &  5  &  6  &  6  &  6  &  6  &  6 \\\\
+1 & 1 & 1 & 1 & 1  &  2  &  3  &  4  &  5  &  6  &  6  &  6  &  6  &  6 \\\\
+1 & 1 & 1 & 1 & 1  &  2  &  3  &  4  &  5  &  6  &  6  &  6  &  6  &  6 \\\\
+1 & 1 & 1 & 1 & \\boxed{1}  &  \\boxed{2}  &  \\boxed{3}  &  \\boxed{4}  &  \\boxed{5}  &  \\boxed{6}  &  6  &  6  &  6  &  6 \\\\
+2 & 2 & 2 & 2 & \\boxed{2}  &  \\boxed{4}  &  \\boxed{6}  &  \\boxed{8}  & \\boxed{10}  & \\boxed{12}  & 12  & 12  & 12  & 12 \\\\
+3 & 3 & 3 & 3 & \\boxed{3}  &  \\boxed{6}  &  \\boxed{9}  & \\boxed{12}  & \\boxed{15}  & \\boxed{18}  & 18  & 18  & 18  & 18 \\\\
+4 & 4 & 4 & 4 & \\boxed{4}  &  \\boxed{8}  & \\boxed{12}  & \\boxed{16}  & \\boxed{20}  & \\boxed{24}  & 24  & 24  & 24  & 24 \\\\
+5 & 5 & 5 & 5 & \\boxed{5}  & \\boxed{10}  & \\boxed{15}  & \\boxed{20}  & \\boxed{25}  & \\boxed{30}  & 30  & 30  & 30  & 30 \\\\
+6 & 6 & 6 & 6 & \\boxed{6}  & \\boxed{12}  & \\boxed{18}  & \\boxed{24}  & \\boxed{30}  & \\boxed{36}  & 36  & 36  & 36  & 36 \\\\
+6 & 6 & 6 & 6 & 6  & 12  & 18  & 24  & 30  & 36  & 36  & 36  & 36  & 36 \\\\
+6 & 6 & 6 & 6 & 6  & 12  & 18  & 24  & 30  & 36  & 36  & 36  & 36  & 36 \\\\
+6 & 6 & 6 & 6 & 6  & 12  & 18  & 24  & 30  & 36  & 36  & 36  & 36  & 36 \\\\
+6 & 6 & 6 & 6 & 6  & 12  & 18  & 24  & 30  & 36  & 36  & 36  & 36  & 36
```

Comment:
I think the solution to this conundrum is to implement software to render latex to   ascii-art  so that mathematics can be printed in the console. See this [post](https://math.stackexchange.com/questions/149303/software-to-render-formulas-to-ascii-art)  for some examples. 

Here is another [good example](http://asciitex.sourceforge.net/). 

I have started working on a Julia package to do this type of thing. If I works out, then perhaps it can be incorporated into Base.  

In the meantime, I have adjusted the indentation of the matrices in the docstring so that when they are printed in the console they align well and are more visually pleasing.  You should find the latest commit more readable. 

Thanks for posting the question on discourse.  There is a precedent for having Latex in docstring in Base. See `?fft` for example.